features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/networksettled/NetworkSettledRequestIdentificationStrategy.kt

```diff
@@ -0,0 +1,19 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric.networksettled
+
+import java.util.concurrent.TimeUnit
+
+internal sealed class NetworkSettledRequestIdentificationStrategy {
+    internal class TimeIntervalBasedStrategy(
+        val timeThresholdNs: Long = TimeUnit.MILLISECONDS.toNanos(DEFAULT_TIME_THRESHOLD_MS)
+    ) : NetworkSettledRequestIdentificationStrategy() {
+        companion object {
+            internal const val DEFAULT_TIME_THRESHOLD_MS = 100L
+        }
+    }
+}
```

Comment:
This is not the end of it, we can discuss a bit more about this if you want. There will be another strategy which will accept a configurable interface which will be the `RequestValidator` which is an interface. I did not expose yet this interface because this will be in another task. The reason why I like strategies is that they are exhaustive and also the client will be able to pick from a definite set. One strategy is based only on a time interval so the above definition doesn't match. The client will have to only define the `timeInterval` and that will be it he doesn't need to implement any interface. The default value is `100ms` as this is defined in the RFC. I guess you will see it better once we add the other tasks also.